components/Table/Table.js

```diff
@@ -95,12 +113,184 @@ export class Table {
+            return;
+          }
+        }
+        return elConfig != null;
+      })
+      .map((menuElement) => {
+        this.menuItem = document.createElement('a');
+        this.menuItem.classList.add('menu-item');
+        this.menuItem.textContent = menuElement.label;
+
+        this.menuItem.addEventListener('click', () => {
+          const dataKey = cell.getAttribute('data-key');
+          switch (menuElement.key) {
+            case MenuElMap.UNSORT:
+              this.callbacksMap.sortCallback();
+              break;
+            case MenuElMap.SORT_BY_ASC:
+              this.isOrderAsc = true;
+              this.callbacksMap.sortCallback(dataKey, this.isOrderAsc);
+              break;
+            case MenuElMap.SORT_BY_DESC:
+              this.isOrderAsc = false;
+              this.callbacksMap.sortCallback(dataKey, this.isOrderAsc);
+              break;
+            case MenuElMap.FILTER:
+              this.createFilterMenu();
+              break;
+            case MenuElMap.HIDE:
+              cell.style.display = 'none';
+              const dataElements = this.tableElementsMap[dataKey];
+              if (dataElements) {
+                this.tableElementsMap[dataKey].forEach(
+                  (item) => (item.style.display = 'none')
+                );
+              }
+              break;
+          }
+        });
+        return this.menuItem;
+      });
+
+    dropDownMenu.append(...this.menuDropDownItems);
+    menuBtn.append(dropDownMenu);
+
+    menuBtn.addEventListener('click', () => {
+      dropDownMenu.classList.toggle('show');
+    });
+  }
+
+  createFilterMenu() {
+    const filterMenuContainer = document.createElement('div');
+    filterMenuContainer.classList.add('filter-menu-container');
+
+    const filterLabel = document.createElement('div');
+    filterLabel.classList.add('filter-label');
+    filterLabel.textContent = 'Name';
+
+    const filterOperator = document.createElement('div');
+    filterOperator.classList.add('filter-operator');
+    filterOperator.textContent = 'equal';
+
+    const filterValue = document.createElement('input');
+    filterValue.classList.add('filter-value');
+    filterValue.placeholder = 'Filter value';
+
+    const btnClean = document.createElement('div');
+    btnClean.classList.add('btn-clean');
+    btnClean.textContent = 'x';
+
+    const btnExit = document.createElement('div');
+    btnExit.classList.add('btn-exit');
+    btnExit.textContent = 'x';
+
+    btnExit.addEventListener('click', () => {
+      filterMenuContainer.remove();
+    });
+
+    if (this.callbacksMap.sortCallback) {
+      btnClean.addEventListener('click', () => {
+        filterValue.value = '';
+        this.callbacksMap.sortCallback();
+      });
+    }
+
+    const filterFunction = (e) => {
+      let filter = e.target.value;
+      this.callbacksMap.filterCallback(filter);
+    };
+    filterValue.addEventListener('input', filterFunction);
```

Comment:
Let's move this function in the listener, since it's small we could handle it there